# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsChild.cpp
# Commit: 94ca5d86b3b0
# Full Hash: 94ca5d86b3b0c398294dd82d44194a5173c5aec6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-01-09 21:39:42
# Regressor Bug: 1600906
# File Overlap Count: 1
# Description:
#   Bug 1600906 - Use scoped enums in IDBCursor. r=dom-workers-and-storage-reviewers,ytausky
#   
#   Depends on D57990
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D57991
# ==============================================================================

diff -r 801f97f67366 -r 94ca5d86b3b0 dom/indexedDB/ActorsChild.cpp
--- a/dom/indexedDB/ActorsChild.cpp	Thu Jan 09 14:58:17 2020 +0000
+++ b/dom/indexedDB/ActorsChild.cpp	Mon Jan 06 14:19:39 2020 +0000
@@ -1278,11 +1278,11 @@
 
 auto GetKeyOperator(const IDBCursor::Direction aDirection) {
   switch (aDirection) {
-    case IDBCursor::NEXT:
-    case IDBCursor::NEXT_UNIQUE:
+    case IDBCursor::Direction::Next:
+    case IDBCursor::Direction::NextUnique:
       return &Key::operator>=;
-    case IDBCursor::PREV:
-    case IDBCursor::PREV_UNIQUE:
+    case IDBCursor::Direction::Prev:
+    case IDBCursor::Direction::PrevUnique:
       return &Key::operator<=;
     default:
       MOZ_CRASH("Should never get here.");
@@ -3457,18 +3457,18 @@
 
   auto& item = mDelayedResponses.front();
   switch (mCursor->GetType()) {
-    case IDBCursor::Type_ObjectStore:
+    case IDBCursor::Type::ObjectStore:
       mCursor->Reset(std::move(item.mKey), std::move(item.mCloneInfo));
       break;
-    case IDBCursor::Type_Index:
+    case IDBCursor::Type::Index:
       mCursor->Reset(std::move(item.mKey), std::move(item.mLocaleAwareKey),
                      std::move(item.mObjectStoreKey),
                      std::move(item.mCloneInfo));
       break;
-    case IDBCursor::Type_ObjectStoreKey:
+    case IDBCursor::Type::ObjectStoreKey:
       mCursor->Reset(std::move(item.mKey));
       break;
-    case IDBCursor::Type_IndexKey:
+    case IDBCursor::Type::IndexKey:
       mCursor->Reset(std::move(item.mKey), std::move(item.mLocaleAwareKey),
                      std::move(item.mObjectStoreKey));
       break;